app/services/post_status_service.rb

```diff
@@ -95,6 +107,15 @@ def safeguard_mentions!(status)
     raise UnexpectedMentionsError.new('Post would be sent to unexpected accounts', unexpected_accounts)
   end
 
+  def safeguard_federated_mentions!(status)
```

Comment:
This method will prevent users from mentioning remote users unless the federation level is full federation.